src/order/succ_pred/basic.lean

```diff
@@ -961,6 +961,43 @@ lemma pred.rec_linear {p : α → Prop} (hsucc : ∀ a, p a ↔ p (pred a)) (a b
 end pred_order
 end linear_order
 
+section is_well_order
+variables [linear_order α] [h : is_well_order α (<)] [pred_order α]
+include h
+
+@[priority 100]
+instance is_well_order.to_is_pred_archimedean : is_pred_archimedean α :=
+⟨λ a, begin
+  refine well_founded.fix h.wf (λ b ih hab, _),
+  replace hab := hab.eq_or_lt,
+  rcases hab with rfl | hab,
+  { exact ⟨0, rfl⟩ },
+  cases le_or_lt b (pred b) with hb hb,
+  { have : ∀ {a : α}, ¬ a < b := minimal_of_le_pred hb,
+    exact (this hab).elim },
+  obtain ⟨k, hk⟩ := ih (pred b) hb (le_pred_of_lt hab),
+  refine ⟨k + 1, _⟩,
+  rw [iterate_add_apply, iterate_one, hk],
+end⟩
+
+@[priority 100]
+instance is_well_order.to_is_succ_archimedean [succ_order α] : is_succ_archimedean α :=
```

Comment:
after writing the `pred` instance, I feel like this instance is wrong - I think(?) we'd want `well_order (>)`, which I doubt is a thing that `mathlib` likes considering ever

Reply:
and I can't seem to find the `well_order` instance for fintypes, which should be automatic

Reply:
That's okay. I've used `(>)` in instances before. In that case, it's just a shorthand for `(swap (<))`. Indeed we seem to be missing the `fintype +linear_order -> well-order` instances.

Reply:
hmm, we have [`preorder.well_founded`](https://leanprover-community.github.io/mathlib_docs/data/fintype/basic.html#fintype.preorder.well_founded) and the upgrading to linear order after that; can we make an instance `is_well_order` for any relation with the right attributes on a `fintype`?

Reply:
what's your thoughts on this @YaelDillies? I'm just bumping this to use the new stuff right now, but this is still an unresolved issue. How should we state this new lemma? do we have the right instances on (>) already to make any preorder-like relation work with (>)?

Reply:
I think it should work fine.